File: c++ applications/Simulator/JNASimulatorLink.cpp
#include "JNASimulatorLink.h"

double getTime(){
    return simulator->getTime();
}

void performOneStepOfSimulation(){
    simulator->performOneStepOfSimulation();
}

void performWholeSimulation(){
    simulator->performWholeSimulation();
}

void setSimulatorForNewSimulation(int seed, char* settingsPath){
    delete simulator;
    fstream settingsFile;
    settingsFile.open(settingsPath);
    if(!settingsFile){
        cerr << "Could not load settings file";
        exit(1);
    }
    double Rc, Rd,closeTime;
    settingsFile >> Rc;
    settingsFile >> Rd;
    settingsFile >> closeTime;
    settingsFile.close();
    simulator = new DiscreteEventSimulator(seed,Rc,Rd,closeTime);
}

double rval(int obs){
    return simulator->rval(obs);
}

double rvals(char* obs){
    return simulator->rval(string(obs));
}

File: Multivesta meets c++/src/SimpleBank/JNASimulatorLink.java
package SimpleBank;

import com.sun.jna.Library;
import com.sun.jna.Native;

public interface JNASimulatorLink extends Library {
    JNASimulatorLink INSTANCE = (JNASimulatorLink) Native.load("Simulator", JNASimulatorLink.class);

    double getTime();
    void performOneStepOfSimulation();
    void performWholeSimulation();
    void setSimulatorForNewSimulation(int seed, String settingsPath);
    double rval(int obs);
    double rvals(String obs);
}
